### ISSUE TITLE
feat(core): add specifier type for * & latest

### ISSUE THREAD

**Issue by wtchnm** [JamieMason/syncpack#174]
## Description

Hello, thank you for this great tool! I have a question about the usage of the `*` protocol. While it's nice for `peerDependencies`, I find it less ideal in `devDependencies` where I prefer to have an exact version.

## Suggested Solution

I propose the following configuration to support banning a specific version of a package, serving as a potential security measure:
```json
"versionGroups": [
  {
    "dependencyTypes": ["dev"],
    "pinVersion": "*",
    "isBanned": true
  }
]
```
This would allow us to explicitly disallow the use of the `*` protocol in devDependencies.

## Help Needed

I am willing to create a PR if the suggested solution aligns with the project's goals.

**Comment by JamieMason** [comment]
Thanks a lot @wtchnm, this sounds like it could be addressed by something myself and @moltar were discussing [here](https://github.com/JamieMason/syncpack/issues/163#issuecomment-1735341148) – please take a look and see if you agree with that approach 👍

**Comment by wtchnm** [comment]
Sorry @JamieMason, I searched the repo for similar issues but totally missed that one. I agree with the solution you proposed!

**Comment by JamieMason** [comment]
No problem at all 👍

**Comment by JamieMason** [comment]
Released in [12.2.0](https://github.com/JamieMason/syncpack/releases/tag/12.2.0).

👋 If anyone reading this finds syncpack useful, please tell people about it.

**Comment by wtchnm** [comment]
Thanks @JamieMason, it worked great with `specifierTypes`:

```
"versionGroups": [
    {
      "label": "Ban * protocol in devDependencies",
      "dependencyTypes": ["dev"],
      "specifierTypes": ["latest"],
      "isBanned": true
    }
]
```

I made a PR to fix the examples in the documentation: https://github.com/JamieMason/syncpack/pull/185.
